plugins/inputs/system/memory.go

```diff
@@ -1,14 +1,21 @@
 package system
 
 import (
+	"bufio"
 	"fmt"
+	"os"
+	"strconv"
+	"strings"
 
 	"github.com/influxdata/telegraf"
 	"github.com/influxdata/telegraf/plugins/inputs"
 )
 
+var linuxMeminfoPath = "/proc/meminfo" // so we can override it in tests
+
 type MemStats struct {
-	ps PS
+	ps           PS
+	LinuxMeminfo bool
```

Comment:
Ok, makes sense. Then we need to make it clear in the doc that for linux distros that option should be true to get a more detailed meminfo data.